bson/src/main/org/bson/codecs/pojo/DiscriminatorLookup.java

```diff
@@ -57,7 +61,12 @@ public Class<?> lookup(final String discriminator) {
 
     void addClassModel(final ClassModel<?> classModel) {
         if (classModel.getDiscriminator() != null) {
-            discriminatorClassMap.put(classModel.getDiscriminator(), classModel.getType());
+            Discriminator discriminator = Discriminator.create(classModel);
+            Class<?> cachedClass = discriminatorClassMap.computeIfAbsent(discriminator, (cachedDiscriminator) -> classModel.getType());
+            if (cachedClass != classModel.getType()) {
+                LOGGER.error(format("Duplicate %s registered for both: `%s` and `%s`",
```

Comment:
Opted to not throw an exception as this may be discovered during runtime. 

I'm not 100% sure if throwing an exception would be considered a breaking change. In theory if a user was just writing data then there is potential that the code reading the data could be configured differently (multiple PojoCodecProviders) to allow for conflict free reading. 🤷 

So opted to log an error.

cc @jyemin - any thoughts?